### PR TITLE
[tckrefactor] Jakarta Persistence 3.2 - initial commit (java.time.Instant and java.time.Year)

### PR DESCRIPTION
**Pull request by rfelcman**
**Describe the change**
This is initial PR for upcoming Jakarta Persistence 3.2 specification. Second version of #1230 to adopt Platform TCK refactor changes
List if changes:

- jakarta.persistence-api dependency change 3.2.0-M2 (currently available in the Jakarta staging)
- Test refactor to integrate `ee.jakarta.tck.persistence.jpa22` into other tests. Started with `jpa22/datetime`
- Initial test change in `ee.jakarta.tck.persistence.core.types.datetime` to verify new `@Basic` types `java.time.Instant` and `java.time.Year`

**Additional context**
Jakarta Persistence 3.2.0 spec - new features
https://projects.eclipse.org/projects/ee4j.jpa/releases/3.2
https://jakartaee.github.io/persistence/latest-nightly/nightly.html#jakarta-persistence-3-2

CC @lukasj @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

File: jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/types/datetime/DateTimeEntity.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.NamedQuery;
+import jakarta.persistence.Table;
+
+/**
+ * Entity containing new JPA date and time types.
+ */
+@Entity
+@Table(name = "DATATYPES_DATE_TIME")
```
Comment (scottmarlow) [inline jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/types/datetime/DateTimeEntity.java:38]: Are we automatically creating the DATATYPES_DATE_TIME table somewhere?

Comment (scottmarlow) [reply jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/types/datetime/DateTimeEntity.java:38]: I guess so via https://github.com/jakartaee/platform-tck/pull/1240/files#diff-e4a21611b5e0c0ce1eceb4929a0fcf820d562e6fc17ef810fa168a9def899cfcR69 :)

**Comment by scottmarlow** [comment]
CC @lukasj

File: jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/types/datetime/DateTimeEntity.java (hunk 1 above)
Comment (rfelcman) [reply jpa/spec-tests/src/main/java/ee/jakarta/tck/persistence/core/types/datetime/DateTimeEntity.java:38]: > I guess so via https://github.com/jakartaee/platform-tck/pull/1240/files#diff-e4a21611b5e0c0ce1eceb4929a0fcf820d562e6fc17ef810fa168a9def899cfcR69 :)

Yes, that is right persistence property.

**Review by lukasj — APPROVED** [review]

**Review by scottmarlow — APPROVED** [review]

**Comment by scottmarlow** [comment]
Thanks @rfelcman for the change!

Thanks @lukasj for reviewing!

**Pull request merged**